src/core/tsi/ssl_transport_security.c

```diff
@@ -1013,6 +1062,8 @@ static tsi_result ssl_handshaker_create_frame_protector(
 static void ssl_handshaker_destroy(tsi_handshaker *self) {
   tsi_ssl_handshaker *impl = (tsi_ssl_handshaker *)self;
   SSL_free(impl->ssl); /* The BIO objects are owned by ssl */
+  tsi_ssl_handshaker_factory_unref(impl->factory_ref);
+  impl->factory_ref = NULL;
```

Comment:
You don't even need this... sorry.

Reply:
That is the result of many hours lost in front of debuggers chasing down zombie references. Ie. you can't accidentally dereference something that isn't there...

Anyway, I'll remove it.